src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallAppsTask.groovy

```diff
@@ -142,4 +142,154 @@ class InstallAppsTask extends AbstractServerTask {
+            config.toXmlFile(looseConfigFile)
+            break
+        case "ear":
+            break
+        default:
+            logger.info(MessageFormat.format(("Loose application configuration is not supported for packaging type {0}. The project artifact will be installed as is."),
+                    project.getPackaging()));
+            installProjectArchive()
+            break
+        }
+    }
+
+    protected void installLooseConfigWar(LooseConfigData config) throws Exception {
+        File dir = getServerDir(project)
+        if (!dir.exists() && containsJavaSource()) {
+          throw new GradleException("Failed to install loose application from project {0}. The project has not been compiled.", project.name)
+        }
+        LooseWarApplication looseWar = new LooseWarApplication(project, config)
+        looseWar.addSourceDir()
+        looseWar.addOutputDir(looseWar.getDocumentRoot(), project, "/WEB-INF/classes");
+
+        //retrieves dependent library jar files
+        addWarEmbeddedLib(looseWar.getDocumentRoot(), looseWar);
+
+        //add Manifest file
+        File manifestFile = new File(project.sourceSets.main.getOutput().getResourcesDir().getParentFile().getAbsolutePath() + "/META-INF/MANIFEST.MF")
+        looseWar.addManifestFile(manifestFile, "gradle-war-plugin")
+    }
+
+    private void addWarEmbeddedLib(Element parent, LooseApplication looseApp) throws Exception {
+      ArrayList<File> deps = new ArrayList<File>();
+      project.configurations.runtime.each { deps.add(it)}
+      project.configurations.providedRuntime.each { deps.remove(it) }
+      File parentProjectDir = new File(project.getRootProject().rootDir.getAbsolutePath())
+      for (File dep: deps) {
+        String projectPath = getProjectPath(parentProjectDir, dep)
+        if(!projectPath.isEmpty() && project.getRootProject().findProject(projectPath) != null){
+            Element archive = looseApp.addArchive(parent, "/WEB-INF/lib/"+ dep.getName());
+            looseApp.addOutputDirectory(archive, project.getRootProject().findProject(projectPath), "/");
+            looseApp.addManifestFile(archive, project.getRootProject().findProject(projectPath), "gradle-jar-plugin");
+        } else{
+            looseApp.getConfig().addFile(parent, dep.getAbsolutePath() , "/WEB-INF/lib/" + dep.getName());
+        }
+      }
+    }
+
+    private String getProjectPath(File parentProjectDir, File dep){
+      String projectJunk = dep.getAbsolutePath().replace(parentProjectDir.getAbsolutePath()+"/","")
+      String projectPath = dep.getAbsolutePath().replace(projectJunk,"")
+      Pattern pattern = Pattern.compile("/build/.*")
+      Matcher matcher = pattern.matcher(projectJunk)
+      projectPath = matcher.replaceAll("")
+      return projectPath;
+    }
+
+    private boolean containsJavaSource(){
+      Set<File> srcDirs = project.sourceSets.allJava.getSrcDirs();
```

Comment:
https://docs.gradle.org/4.1/dsl/org.gradle.api.tasks.SourceSet.html
if you get allJava or java from the source set, can you just check if the list is empty or not?
is it just a collection of .java files or folders as well?